### DISCUSSION TITLE
How to get mermaid diagrams working with readthedocs template?

### DISCUSSION THREAD

**Discussion by ericthomas1** [mkdocs/mkdocs#3101] [Q&A] [answered]
I see there is a `mermaid` configuration referenced in the `mkdocs-material` theme...
![image](https://user-images.githubusercontent.com/42788729/211898120-dc7dac41-a580-4c1f-a447-22ed4711ee91.png)

I don't see a `mermaid` config referenced in the default theme...
![image](https://user-images.githubusercontent.com/42788729/211898202-e906b5eb-9ed2-4c60-ad6f-d555dfb8e0c4.png)

How do I get `mermaid` working with the `readthedocs` theme?

**Comment by Andre601** [comment]
Mermaid support is only part of Material for MkDocs which isn't a default theme in MkDocs itself.

**Comment by facelessuser** [comment, marked as answer]
With a little JS, maybe some CSS, and a custom fence using SuperFences, you can probably add it.

Here's a CodePen using a custom JS loader and creating a JS config for Mermaid to transform a code block into a diagram. https://codepen.io/facelessuser/pen/oNeNydQ

Here explains how you'd setup your custom fence using SuperFences in MkDocs: https://facelessuser.github.io/pymdown-extensions/extras/mermaid/#using-in-mkdocs

**Reply by ericthomas1** [reply]
Oh wow+! That SuperFences link is great. Thank you.

**Reply by ericthomas1** [reply]
I was able to get the example to work. Do you know how I can control the scale of `mermaid` diagrams so they fit relative to the rest of the docs? 

![image](https://user-images.githubusercontent.com/42788729/211936812-46ecf57f-ed25-40fd-b708-0c97d5ab91ee.png)

**Reply by facelessuser** [reply]
You can do 1 of 2 things.

1. Use CSS to make the content scrollable; this keeps the image a consistent size, but allows you to scroll images that are larger than the viewable width. 
2. Or you can set `useMaxWidth` in the Mermaid config to `True`.

I cover this in my Mermaid notes: https://facelessuser.github.io/pymdown-extensions/extras/mermaid/#configuration.

**Reply by ericthomas1** [reply]
Thank you. I went your second suggestion. This looks good to me (so far). I can see it potentially creating wide diagrams that are too small.

Speaking of, is it advisable to increase the width of the viewable screen? I have a lot of dead space to use. 

![image](https://user-images.githubusercontent.com/42788729/212159978-3e830f08-671f-48ae-81b7-a596297c61d8.png)

**Reply by facelessuser** [reply]
🤷🏻 That's just how the "Read the Docs" theme has always looked. I've never cared for their approach in that regard. It's not uncommon for a site to not use the full width of your screen to display text to make it easier to read, but usually, sites will center the main content and try to make the sides blend in more than having a large gray "dead zone". You are certainly free to add additional CSS to modify the theme to suite your tastes and/or needs.

**Reply by ericthomas1** [reply]
Right on. I found the appropriate class and added it to `docs/css/extra.css`.

**For posterity**: I changed `max-width` from `800px` to `100%`.

```
.wy-nav-content {
    max-width: 100%;
}
```
**Much better:**

![image](https://user-images.githubusercontent.com/42788729/212162910-b6c1783f-32e0-4633-8296-8f80b6b4f16c.png)
